base/base/cgroupsv2.cpp

```diff
@@ -9,11 +9,18 @@
 bool cgroupsV2Enabled()
 {
 #if defined(OS_LINUX)
-    /// This file exists iff the host has cgroups v2 enabled.
-    auto controllers_file = default_cgroups_mount / "cgroup.controllers";
-    if (!std::filesystem::exists(controllers_file))
-        return false;
-    return true;
+    try
+    {
+        /// This file exists iff the host has cgroups v2 enabled.
+        auto controllers_file = default_cgroups_mount / "cgroup.controllers";
+        if (!std::filesystem::exists(controllers_file))
+            return false;
+        return true;
+    }
+    catch (...)
+    {
+        return false; /// e.g. permission denied exception
```

Comment:
May be we should explicitly catch std::filesystem error and check for error code permission denied and only in this case return false and otherwise throw?

Reply:
You are right that catching `...` is too coarse.

`fs::filesystem::exists` throws ()
- `std::filesystem::filesystem_error` for "underlying API issues" ([docs](https://en.cppreference.com/w/cpp/filesystem/exists)) e.g. wrong permissions). We should catch and ignore any of these _without logging_ as most callers provide a fallback to cgroups v1.
- `std::bad_alloc` for OOMs and `std::exception` for everything else. We should let these simply bubble up.